frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -64,20 +65,90 @@ local function tidy_markup(results)
     return results
 end
 
+function ReaderDictionary:cleanSelection(text, is_single_word)
+    -- Will be used by ReaderWikipedia too (with is_single_word false)
+    -- We do multiple times the same replacements, which is most of the time overkill,
+    -- but sometimes provices better cleaning
```

Comment:
Yes, adding samples in comment should be fine for now. I can help convert them to uni tests later, it's actually very easy to add one. I am more concerned whether doing multiple passes of replacements is the correct approach here, which can be verified with examples. In theory, one pass should be enough to extract the needed word with lpeg.

Yes, it's fine to have language specific rules here.